src/org/labkey/test/tests/SampleTypeLinkToStudyTest.java

```diff
@@ -517,11 +564,52 @@ private void linkToStudy(String targetStudy, String sampleName, int numOfRowsToB
 
         log("Link to study: Choose target");
         selectOptionByText(Locator.id("targetStudy"), "/" + targetStudy + " (" + targetStudy + " Study)");
+        if (categoryName != null)
+            setFormElement(Locator.name("autoLinkCategory"), categoryName);
         clickButton("Next");
 
         new DataRegionTable("query", getDriver()).clickHeaderButtonAndWait("Link to Study");
     }
 
+    private void createDatasetCategory(String projectName, String name)
```

Comment:
I'm a little surprised there's not a pre-existing helper that makes dataset categories! If you think it'd be easy and useful, I'd consider pulling this into a shareable place so others could reuse the dataset category creation function. No big deal if you think that that wouldn't be worth it, though.

Reply:
Could you use the `org.labkey.remoteapi.reports.Category` class?
It looks like it might do what you need.